mllib/src/main/scala/org/apache/spark/mllib/tree/DecisionTree.scala

```diff
@@ -62,8 +64,9 @@ class DecisionTree private[spark] (private val strategy: Strategy, private val s
    */
   @Since("1.2.0")
   def run(input: RDD[LabeledPoint]): DecisionTreeModel = {
-    val rf = new RandomForest(strategy, numTrees = 1, featureSubsetStrategy = "all",
-      seed = seed)
+    val instr = Instrumentation.create(new DecisionTreeClassifier, input)
+    val rf =
+      new RandomForest(strategy, numTrees = 1, featureSubsetStrategy = "all", seed = seed, instr)
```

Comment:
Yes, the instrumentation API was designed with the pipeline API in mind. I suggest we make the instrumentation argument an `Option[Instrumentation[_]]`, and not use it in tests and mllib.